mistakedetection/src/ca/mcgill/sel/mistakedetection/MistakeDetection.java

```diff
@@ -3304,10 +3320,42 @@ public static String taggerOut(String taggerInput) {
     return maxentTagger.tagString(taggerInput);
   }
 
-  private static boolean spellingMistakeCheck(String name1, String name2) {
-    int lDistance = levenshteinDistance(name1, name2);
-    return lDistance > 0 && lDistance <= MAX_LEVENSHTEIN_DISTANCE_ALLOWED;
+  private static boolean isSpelledIncorrectly(int lDistance, int studStringLength, int instStringLength) {
+ 
+    if( lDistance == 0) {
+      return false;
+    }
+    int limit = 2;
+    if (instStringLength > 5 || studStringLength > 5) {
+      return lDistance <= limit;
+    } else if (instStringLength > 2 || studStringLength > 2) {
+      limit = 1;
+      return lDistance <= limit;
+    } else {
+      limit = 0;
+      return lDistance == limit;
+    }
   }
+  
+  private static boolean isStringWrong(String studString, String instString) {
+    int lDistance = levenshteinDistance(studString, instString);
+    if( lDistance == 0) {
+      return false;
+    }
+    int limit = 2;
+    int instStringLength = instString.length();
+    int studStringLength = studString.length();
+    if (instStringLength > 5 || studStringLength > 5) {
```

Comment:
The logic here is repeated above, extract it to avoid repetition

Reply:
actually the if conditions are same but return statements are diff